tools/install.sh

```diff
@@ -90,12 +92,12 @@ main() {
   TEST_CURRENT_SHELL=$(expr "$SHELL" : '.*/\(.*\)')
   if [ "$TEST_CURRENT_SHELL" != "zsh" ]; then
     # If this platform provides a "chsh" command (not Cygwin), do it, man!
-    if hash chsh >/dev/null 2>&1; then
+    if [ -f /etc/shells ] && [ hash chsh >/dev/null 2>&1 ]; then
```

Comment:
I don't think you want the `[ ... ]` around the `hash` command in the second part of this line. It turns it into a broken textual condition test, and the `2>&1` inside the test is hiding the error message that results.

```
sh-3.2$ if [ hash chsh >/dev/null 2>&1 ]; then echo yup; fi
sh-3.2$ if [ hash chsh >/dev/null  ]; then echo yup; fi
sh: [: hash: unary operator expected
sh-3.2$
```